### DISCUSSION TITLE
Directly call zitadel endpoints or create intermediary enpoints?

### DISCUSSION THREAD

**Discussion by Susheelkaram** [zitadel/zitadel#8802] [Q&A] [answered]
What is general good practice followed for self-hosted zitadel. Do you call the zitadel auth apis directly from client (Mobile app or web) or should I write endpoints which internally call zitadel apis?
Or is it a mix of both ?

**Comment by fforootd** [comment, marked as answer]
Its is typically a good idea to seperate two groups of operations:

1) Calling an api with the current logged in user to fetch/update details of a single user can be done from a client withouth problems even when located with the user (SPA, native apps,..)
2) Calling privileged apis (creating new users, adding permissions, ...) should be done from a server side code.

**Reply by Susheelkaram** [reply]
Thanks for the reply. That sounds about right.
